internal/testutil/log.go

```diff
@@ -0,0 +1,91 @@
+package testutil
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/go-logr/logr"
+)
+
+// Logger implements logr.Logger and logs to testing.T to preserve the order of log lines in tests.
+type Logger struct {
```

Comment:
Remove this and use https://pkg.go.dev/github.com/go-logr/logr@v1.2.3/testr instead. logr already provides coupling with testing